### PR TITLE
Add ClusterID to alert message

### PR DESCRIPTION
**Pull request by NautiluX**
* Simplify login flow from alert to cluster
* ClusterID is read from ClusterVersion

### CODE REVIEW

**Comment by RiRa12621** [comment]
/lgtm
Maybe the unrelated logging improvements should have been a separate commit tho

**Comment by jewzaam** [comment]
Similar to https://github.com/openshift/configure-alertmanager-operator/pull/99 and expect some conflicts, please work with @cblecker on how you want to merge these.

**Comment by cblecker** [comment]
@NautiluX #99 has merged. You're clear to rebase :)

**Comment by NautiluX** [comment]
rebased to master

**Comment by NautiluX** [comment]
@jewzaam I think we can cancel the hold now if you agree.

File: pkg/controller/secret/secret_controller.go [hunk 1]
```diff
@@ -433,6 +436,16 @@ func (r *ReconcileSecret) Reconcile(request reconcile.Request) (reconcile.Result
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileSecret) getClusterID() string {
+	var version configv1.ClusterVersion
+	err := r.client.Get(context.TODO(), client.ObjectKey{Name: "version"}, &version)
+	if err != nil {
+		log.Error(err, "Error reading cluster id.")
```
Comment (cblecker) [inline pkg/controller/secret/secret_controller.go:443]: Can we return this as an `error`, and then log it in the main loop (with the log with variables)?

File: pkg/controller/secret/secret_controller_test.go [hunk 2]
```diff
@@ -297,13 +299,13 @@ func Test_createPagerdutyRoute(t *testing.T) {
 }
 
 func Test_createPagerdutyReceivers_WithoutKey(t *testing.T) {
-	assertEquals(t, 0, len(createPagerdutyReceivers("", "")), "Number of Receivers")
+	assertEquals(t, 0, len(createPagerdutyReceivers("", "", "fake-cluster-id")), "Number of Receivers")
```
Comment (cblecker) [inline pkg/controller/secret/secret_controller_test.go:302]: In this instance, all three strings should be empty

File: pkg/controller/secret/secret_controller_test.go [hunk 3]
```diff
@@ -297,13 +299,13 @@ func Test_createPagerdutyRoute(t *testing.T) {
...
 }
 
 func Test_createPagerdutyReceivers_WithKey(t *testing.T) {
 	key := "abcdefg1234567890"
 
-	receivers := createPagerdutyReceivers(key, exampleConsoleUrl)
+	receivers := createPagerdutyReceivers(key, exampleConsoleUrl, "fake-cluster-id")
```
Comment (cblecker) [inline pkg/controller/secret/secret_controller_test.go:308]: Can we make `"fake-cluster-id"` a const, and then use it in all these places? This just ensures that the test and the example string always match

File: pkg/controller/secret/secret_controller_test.go [hunk 4]
```diff
@@ -630,7 +651,7 @@ func Test_ReconcileSecrets(t *testing.T) {
 
 		// Create the secrets for this specific test.
 		if tt.amExists {
-			writeAlertManagerConfig(reconciler, createAlertManagerConfig("", "", ""))
+			writeAlertManagerConfig(reconciler, createAlertManagerConfig("", "", "", "fake-cluster-id"))
```
Comment (cblecker) [inline pkg/controller/secret/secret_controller_test.go:654]: This should also be a blank string (the test confirms that the blank string is overwritten)

File: pkg/controller/secret/secret_controller_test.go [hunk 5]
```diff
@@ -644,10 +665,11 @@ func Test_ReconcileSecrets(t *testing.T) {
 			createSecret(reconciler, secretNamePD, secretKeyPD, pdKey)
 		}
 
-		configExpected := createAlertManagerConfig(pdKey, wdURL, exampleConsoleUrl)
+		configExpected := createAlertManagerConfig(pdKey, wdURL, exampleConsoleUrl, "fake-cluster-id")
 
 		verifyInhibitRules(t, configExpected.InhibitRules)
 
+		createClusterVersion(reconciler)
```
Comment (cblecker) [inline pkg/controller/secret/secret_controller_test.go:672]: I think this needs to happen sooner, doesn't it? Like, line 647?

File: pkg/controller/secret/secret_controller.go (hunk 1 above)
Comment (NautiluX) [reply pkg/controller/secret/secret_controller.go:443]: Sure, I don't have any preference about where we hide the error ;)

File: pkg/controller/secret/secret_controller_test.go (hunk 2 above)
Comment (NautiluX) [reply pkg/controller/secret/secret_controller_test.go:302]: done

File: pkg/controller/secret/secret_controller_test.go (hunk 4 above)
Comment (NautiluX) [reply pkg/controller/secret/secret_controller_test.go:654]: make sense. Done

File: pkg/controller/secret/secret_controller_test.go (hunk 5 above)
Comment (NautiluX) [reply pkg/controller/secret/secret_controller_test.go:672]: yes, now that the test actually does something I need to move it :)

**Comment by NautiluX** [comment]
Thanks for your review @cblecker, addressed all your comments in the latest commit.

**Pull request merged**